pydantic/_analyzed_type.py

```diff
@@ -0,0 +1,213 @@
+from __future__ import annotations
+
+import sys
+from typing import TYPE_CHECKING, Any, Dict, Generic, Iterable, Set, TypeVar, Union, overload
+
+from pydantic_core import CoreConfig, SchemaSerializer, SchemaValidator, core_schema
+from typing_extensions import Literal
+
+from pydantic.json_schema import DEFAULT_REF_TEMPLATE, GenerateJsonSchema
+
+from ._internal import (
+    _generate_schema,
+    _typing_extra,
+)
+
+T = TypeVar('T')
+
+if TYPE_CHECKING:
+    # should be `set[int] | set[str] | dict[int, IncEx] | dict[str, IncEx] | None`, but mypy can't cope
+    IncEx = Union[Set[int], Set[str], Dict[int, Any], Dict[str, Any]]
+
+
+class AnalyzedType(Generic[T]):
+    def __init__(
+        self,
+        core_schema: core_schema.CoreSchema,
```

Comment:
Maybe I'm missing something, but this should definitely take type as the first and only positional argument no?

Reply:
That's what `AnalyzedType.analyze_type` (which I would rename to `.analyze`) is for, more in https://github.com/pydantic/pydantic/pull/5308#discussion_r1154709859

Reply:
I don't agree, if we need a zero overhead type, make it private in the background.

The primary interface for processing (validating and serialising) data that doesn't fit into model, **must** be simple and obvious to use.

Otherwise we risk becoming attrs.

Reply:
We might not be the only ones needing a zero-overhead type. The pattern of keeping `__init__` low overhead and doing extra work in static/class methods is widely used and well understood. But if you really feel that strongly I can try to move this into `__init__`...

Reply:
I'm not opposed to having a public method with zero overhead, but the main interface we expose (along side BaseModel, dataclass, and validate_aeguments) needs to have the signature `Thing(TheType).validate(data)`.

Thanks for understanding.